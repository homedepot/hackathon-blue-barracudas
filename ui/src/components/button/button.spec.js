import React from 'react';
import {mount, shallow} from "enzyme";
import Button from "./button";
import toJson from 'enzyme-to-json';

describe('<Button />', () => {

        let props;
        let wrapper;
        let mountedProps;
        let mountWrapper;

        beforeAll(() => {
            mountedProps = {
                className: 'btnOrdAcc',
                handleAction: jest.fn(),
                label: 'Order Accessory',
                id: 'btnOrdAccessory',
                isPrimary: true
            };
            props = mountedProps;

            wrapper = shallow(<Button {...props}/>);
            mountWrapper = mount(<Button {...props}/>);
        });

        it('should render button component properly', () => {
            expect(toJson(wrapper)).toMatchSnapshot();

        });

        it('should render button with primary and other class name', () => {
            expect(wrapper.find('button.btnOrdAcc.primary')).toHaveLength(1);
        });

        it('should render button with secondary and other class name', () => {
            wrapper.setProps({isPrimary: false});
            expect(wrapper.find('button.btnOrdAcc.secondary')).toHaveLength(1);
        });

        it('should call onClick function on button action', () => {
            const spy = jest.spyOn(mountedProps, "handleAction");
            mountWrapper = mount(<Button {...mountedProps} />);
            mountWrapper.find("button.btnOrdAcc.primary").simulate("click");
            expect(spy).toHaveBeenCalled();
            spy.mockReset();
            spy.mockRestore();
        });
    }
);