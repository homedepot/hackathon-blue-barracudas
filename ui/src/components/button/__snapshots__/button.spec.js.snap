// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button component properly 1`] = `
<span
  className="accessory-btn"
>
  <button
    className="button btnOrdAcc primary"
    data-ga="ButtonClick|Order Accessory"
    id="btnOrdAccessory"
    onClick={[Function]}
  >
    Order Accessory
  </button>
</span>
`;
